### ISSUE TITLE
Form validation: validating onfocusout, WCAG Compliant?

### ISSUE THREAD

**Issue by Karomar** [wet-boew/wet-boew#8160]
Hi, 

I am wondering if the functionality of triggering the validation on the onfocusout event is WCAG compliant and if so where I could find some information or the specific WCAG rules specifying that this is permitted?

The reason I am asking is in our department we have an in-house WCAG group that has indicated to us that this is not WCAG compliant and that all such validation should only be triggered when the user hits the Save button, or navigates away from the page. I am trying to challenge this but am having a hard time finding WCAG documentation to support my argument.

Is there an easy way to turn off the onfocusevent validation trigger?

```javascript

// The jQuery validation plug-in in action
validator = $form.validate( {
    ...
    ...
    /* adds on tab validation */
    onfocusout: function( element ) {
        this.element( element );
    }
} ); /* end of validate() */
```

Thanks!

**Comment by duboisp** [comment]
@Karomar can you describe what specific WCAG Success Criteria it is failling and which techniques and faillures, if any, are applicable to that accessibility issue?

**Comment by rlambert27** [comment]
It would be great to ask your WCAG group to cite a particular success criteria that fails or even a common failure that has been achieved.

WCAG actually does encourage automatic error checking through SC 3.3.1 and this includes a couple of sufficient techniques such as:

G83 - https://www.w3.org/TR/WCAG20-TECHS/G83.html
G85 - https://www.w3.org/TR/WCAG20-TECHS/G85.html
SCR18 - https://www.w3.org/TR/WCAG20-TECHS/SCR18.html
SCR32 - https://www.w3.org/TR/WCAG20-TECHS/SCR32.html

IMO it's not up to you to challenge this requirement but up to your WCAG group to provide more information on why this fails accessibility. The SC 3.3.1 is pretty darn clear that automatic error checking is perfectly acceptable if done properly.

**Comment by Karomar** [comment]
Thanks @rlambert27! , this is exactly the kind of information I was looking for to go back to them with. I agree that it shouldn't be up to me to tell them what is and isn't compliant but unfortunately they are a roadblock in our department for getting applications approved.

I will let you know what they come back with as it seems like this is perfectly acceptable behavior.

**Comment by Karomar** [comment]
Based on the information cited from rlambert27 our WCAG group determined that this is perfectly valid functionality and is indeed WCAG so thanks for you help!
